zerver/lib/onboarding.py

```diff
@@ -244,6 +244,31 @@ def send_welcome_bot_response(send_request: SendMessageRequest) -> None:
     welcome_bot = get_system_bot(settings.WELCOME_BOT, send_request.realm.id)
     human_response_lower = send_request.message.content.lower()
     content = select_welcome_bot_response(human_response_lower)
+    human_recipient_id = send_request.message.sender.recipient_id
+    assert human_recipient_id is not None
+    realm_id = send_request.message.realm_id
+
+    if (
+        Message.objects.filter(
+            realm_id=realm_id,
+            sender_id=send_request.message.sender.id,
+            recipient_id=send_request.message.recipient.id,
+        ).count()
+        == 1
```

Comment:
Verify that it uses `zerver_message_realm_sender_recipient` index using `explain analyze` in dbshell (See Message model for the index). 

Update the comment in the Message model for where the index is used & add a "Uses index 'zerver_message_realm_sender_recipient'" comment here. -- It helps to be cautious while changing query or removing index in the future.

Reply:
Thanks, how can I verify this, I did not understand.

Reply:
You can use `python manage.py dbshell`

Then [explain analyze](https://www.postgresql.org/docs/current/sql-explain.html) to get an idea of the query plan.